### PR TITLE
refactor: move cohort-selection to components

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: src/app/components/cohort-selection/cohort-selection.directive.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+     * the user can create multiple cohort-selection sections.
+     * @memberof transmartBaseUi
+     * @ngdoc directive
+     * @name cohortSelection
+     */
+    .directive('cohortSelection', function () {
+        return {
+            restrict: 'E',
+            templateUrl: 'app/components/cohort-selection/cohort-selection.tpl.html',
+            controller: 'CohortSelectionCtrl as cohortSelectionCtrl',
```
Comment (rnugraha) [inline src/app/components/cohort-selection/cohort-selection.directive.js]: `controller: 'CohortSelectionCtrl as cohortSelectionCtrl',`
this is unnecessary

File: src/app/components/cohort-selection/cohort-selection.directive.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            controllerAs : 'cohortSelectionCtrl',
+            link: function (scope, el, attr) {
```
Comment (rnugraha) [inline src/app/components/cohort-selection/cohort-selection.directive.js]: remove link if there're nothing in it

File: src/app/components/cohort-selection/cohort-selection.directive.spec.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ */
+'use strict';
+
+describe('cohort-selection directive', function () {
+
+    // Load the transmartBaseUi module, which contains the directive
+    beforeEach(function () {
+        module('transmartBaseUi');
+    });
+
```
Comment (rnugraha) [inline src/app/components/cohort-selection/cohort-selection.directive.spec.js:36]: maybe create test at least to test if it render correct result?

File: src/app/components/save-cohort/save-cohort-dialog.controller.js [hunk 4]
```diff
@@ -13,14 +13,19 @@ angular.module('transmartBaseUi')
 
             vm.cohortName = '';
 
+            $scope.$on('saveCohortEvent', function (data) {
+                console.log('catching save cohort event, with data: ', data);
```
Comment (rnugraha) [inline src/app/components/save-cohort/save-cohort-dialog.controller.js]: what's the purpose of this piece of code?

File: src/app/containers/workspace/content/content.controller.spec.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+/**
+ * Created by bo on 9/12/16.
+ */
```
Comment (rnugraha) [inline src/app/containers/workspace/content/content.controller.spec.js]: Could you change this to our standard header. You can change JS template header in your IDEA to 

```
/**
 * Copyright (c) ${YEAR} The Hyve B.V.
 * This code is licensed under the GNU General Public License,
 * version 3, or (at your option) any later version.
 */
```

**Comment by kretep** [comment]
I think it might be a good idea to move all of the functionality from the ChartService into the cohort selection component. I think it would be best if each component keeps track of its own charts instead of keeping it in the ChartService. To me, the current structure reminds me of global variables too much and the whole keeping track of the components in a list, using the boxId as an identifier, is only complicating things.

**Comment by beaugogh** [comment]
@kretep yes, actually I am doing it

**Review by kretep — APPROVED** [review]
I approve

**Pull request merged**